packages/babel-plugin/src/utils/css-builders.tsx

```diff
@@ -55,17 +112,17 @@ const extractObjectExpression = (node: t.ObjectExpression, meta: Metadata): CSSO
       } else if (t.isNumericLiteral(propValue)) {
         // We've found a numeric literal like: `fontSize: 12`
         value = addUnitIfNeeded(key, propValue.value);
-      } else if (t.isObjectExpression(propValue)) {
-        // We've found a nested object like: `':hover': { color: 'red' }`
-        const result = extractObjectExpression(propValue, updatedMeta);
-        css += `${key} { ${result.css} }`;
-        variables = variables.concat(result.variables);
+      } else if (t.isObjectExpression(propValue) || t.isLogicalExpression(propValue)) {
+        const result = buildCss(propValue, updatedMeta);
+        css.push(...result.css.map((x) => ({ ...x, css: `${key} { ${getItemCss(x)} }` })));
+        variables.push(...result.variables);
         return;
       } else if (t.isTemplateLiteral(propValue)) {
         // We've found a template literal like: `fontSize: `${fontSize}px`
         const result = extractTemplateLiteral(propValue, updatedMeta);
-        value = result.css;
-        variables = variables.concat(result.variables);
+        css.push(...result.css.map((x) => ({ ...x, css: `${kebabCase(key)}: ${getItemCss(x)};` })));
```

Comment:
nit: lot of places now have `map` calls now along with `getItem` inside it. Can we extract it out if possible. No worries if its not possible atm. We can leave it.

Reply:
the get item css was really only needed because i started investigating the ternary implementation and then needed that because the types were slightly different

this one is slightly different because this is a css declaration - whereas the other is a css rule 🤔 

we could make two functions.. let me have a play